components/director/internal/formationmapping/handler.go

```diff
@@ -144,6 +153,24 @@ func (h *Handler) UpdateFormationAssignmentStatus(w http.ResponseWriter, r *http
 		return
 	}
 
+	if reset {
+		if fa.State != string(model.ReadyFormationState) {
+			log.C(ctx).Error(err)
+			respondWithError(ctx, w, http.StatusBadRequest, errResp)
+			return
+		}
+		reverseFA, err := h.faService.GetReverseBySourceAndTarget(ctx, fa.FormationID, fa.Source, fa.Target)
+		if err != nil {
+			log.C(ctx).Error(err)
+			respondWithError(ctx, w, http.StatusBadRequest, errResp)
```

Comment:
I was imagining it more along the lines of - the reverse is missing, therefore it's a bad request from the client, as it's requesting a reset operation on a formation assignment pair that doesn't have both participants. But if it's a DB error other than Not Found, then it should be Internal Server Error.

Reply:
okay. About the `errResp` it's somewhat "generic" and most of the time is used in case of internal server error. Maybe for the "bad request" errors we could return a more descriptive error/message to the client?
Valid also for the "is not found" if statement and the reverse assignment if.